### PR TITLE
Change the operation name of quartz-scheduler plugin

### PR DESCRIPTION
**Pull request by hailin0**
### Fix <through operationName stats quartz job execution>
- [x] Add a unit test to verify that the fix works.
- [x] Explain briefly about why the bug exists and how to fix it.
  - Use static class name for analysis instead of dynamic jobName
  - Record job execute exception
- reference [quartz-scheduler plugin commit](https://github.com/apache/skywalking/pull/5413)

- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/CHANGES.md).

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ Release Notes.
 * Chore: adapt `create_source_release.sh` to make it runnable on Linux.
 
 #### Java Agent
-
+* Update operationName of quartz-scheduler plugin.
```
Comment (wu-sheng) [inline CHANGES.md:11]: Is this description and PR title correct? The codes seem to add a new tag only. Please confirm.

Comment (hailin0) [reply CHANGES.md:11]: updated [operationName](https://github.com/apache/skywalking/pull/5934/files#diff-818769a4ff74a286eae8711cb67b36f71145fca4d6ebbcaed32d16d0a93181b7R54)

Comment (wu-sheng) [reply CHANGES.md:11]: Got it.

File: apm-sniffer/optional-plugins/quartz-scheduler-2.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/quartz/JobRunShellMethodInterceptor.java [hunk 2]
```diff
@@ -48,14 +49,15 @@
     public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes, MethodInterceptResult result) throws Throwable {
         JobDetail jobDetail = (JobDetail) objInst.getSkyWalkingDynamicField();
 
-        String jobName = jobDetail.getKey().getName();
         String jobGroup = jobDetail.getKey().getGroup();
-        String operationName = ComponentsDefine.QUARTZ_SCHEDULER.getName() + "/" + jobName;
+        String jobName = jobDetail.getKey().getName();
+        String operationName = ComponentsDefine.QUARTZ_SCHEDULER.getName() + "/" + jobDetail.getJobClass().getName();
```
Comment (wu-sheng) [inline apm-sniffer/optional-plugins/quartz-scheduler-2.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/quartz/JobRunShellMethodInterceptor.java:54]: This change related to the agent readme doc. Please fix.

Comment (hailin0) [reply apm-sniffer/optional-plugins/quartz-scheduler-2.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/quartz/JobRunShellMethodInterceptor.java:54]: Please check again

Comment (wu-sheng) [inline apm-sniffer/optional-plugins/quartz-scheduler-2.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/quartz/JobRunShellMethodInterceptor.java:54]: Is one class only allowed to have one job on its methods?

File: CHANGES.md [hunk 3]
```diff
@@ -8,6 +8,7 @@ Release Notes.
 * Chore: adapt `create_source_release.sh` to make it runnable on Linux.
 
 #### Java Agent
+* Update quartz-scheduler plugin.
```
Comment (wu-sheng) [inline CHANGES.md:11]: This changelog is not accurate, please update.

File: apm-sniffer/optional-plugins/quartz-scheduler-2.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/quartz/JobRunShellMethodInterceptor.java (hunk 2 above)
Comment (hailin0) [reply apm-sniffer/optional-plugins/quartz-scheduler-2.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/quartz/JobRunShellMethodInterceptor.java:54]: The user's job needs to implement this [interface](https://github.com/quartz-scheduler/quartz/blob/master/quartz-core/src/main/java/org/quartz/Job.java) , and usually only do one thing.

File: CHANGES.md (hunk 3 above)
Comment (hailin0) [reply CHANGES.md:11]: Need to describe the operationName changes and catch exceptions?

e.g. Update quartz-scheduler plugin operationName and catch job exceptions.

Comment (wu-sheng) [reply CHANGES.md:11]: Such as 

> The operation name of quartz-scheduler plugin in xx case, have been changed as the yyy format.

Comment (hailin0) [reply CHANGES.md:11]: Is this appropriate?


> The operation name of quartz-scheduler plugin in logic span analysis case, have been changed as the `quartz-scheduler/${className}` format.

Comment (wu-sheng) [reply CHANGES.md:11]: The operation name of quartz-scheduler plugin, has been changed as the quartz-scheduler/${className} format.

**Review by wu-sheng — APPROVED** [review]
LGTM

**Pull request merged**